### ISSUE TITLE
ESPNet的方法有的同样问题

### ISSUE THREAD

**Issue by day9011** [wenet-e2e/wenet#221]
使用mask[:, :, :-2:2][:, :, :-2:2]得到的降采样mask和pytorch的卷积公式得到的长度不一致。具体例子如下：
lens = torch.LongTensor([[24], [40], [60], [100]]).cpu()
print(compute_conv_length(compute_conv_length(lens, kernel_size=3, stride=2), kernel_size=3, stride=2))
得到结果：tensor([[ 5],
        [ 9],
        [14],
        [24]])
mask = make_mask_by_length(a, lens).unsqueeze(-2)
new_mask = mask[:, :, :-2:2][:, :, :-2:2]
将布尔值转为int型求和输出
print(torch.sum(new_mask.int(), -1))
tensor([[ 6],
        [10],
        [15],
        [24]])

**Comment by robin1001** [comment]
Yes, it's a problem, we just ignore it now since it has no effect in terms of pipeline and accuracy.
